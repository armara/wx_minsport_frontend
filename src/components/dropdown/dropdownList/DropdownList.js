import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'

import Button from 'components/button/Button'

import 'components/dropdown/dropdownList/DropdownList.scss'

const renderItem = (listItemRenderer, item, titleKey) => {
  if (typeof listItemRenderer === 'function') {
    return listItemRenderer(item, item[titleKey])
  }

  if (typeof item[titleKey] === 'function') {
    return item[titleKey](item)
  }

  return item[titleKey] || ''
}

const DropdownList = ({
  items,
  notFoundPlaceholder,
  listItemRenderer,
  titleKey,
  idKey,
  onItemSelect,
  className,
}) => (
  <ul className={classnames([className, 'dropdown-content', 'menu-list'])}>
    {items.length === 0 ? (
      <span className="dropdown-item dropdown-item_empty">
        {notFoundPlaceholder}
      </span>
    ) : (
      items.map(item => (
        <Button
          type="button"
          key={item[idKey]}
          className="dropdown-item"
          onClick={() => {
            onItemSelect(item)
          }}
        >
          {renderItem(listItemRenderer, item, titleKey)}
        </Button>
      ))
    )}
  </ul>
)

DropdownList.defaultProps = {
  onItemSelect: () => {},
  notFoundPlaceholder: 'Пусто',
  listItemRenderer: null,
  className: '',
}

DropdownList.propTypes = {
  className: PropTypes.string,
  titleKey: PropTypes.string.isRequired,
  idKey: PropTypes.string.isRequired,
  notFoundPlaceholder: PropTypes.string,
  items: PropTypes.arrayOf(PropTypes.object).isRequired,
  listItemRenderer: PropTypes.oneOfType([
    PropTypes.func,
    PropTypes.instanceOf(null),
  ]),
  onItemSelect: PropTypes.func,
}

export default DropdownList
