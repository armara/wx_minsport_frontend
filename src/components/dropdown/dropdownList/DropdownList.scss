@import "styles/mixins";

.menu-list {
  max-height: 350px;
  overflow-y: auto;

  .dropdown-item {
    @include table-text($small, normal, $evil-gray);
    padding: 6px 16px;
    white-space: pre-wrap;
    background-color: unset;
    border: none;
    height: auto;
    cursor: pointer;
  }

  .dropdown-item_empty {
    cursor: default;
    font-style: italic;
  }
}