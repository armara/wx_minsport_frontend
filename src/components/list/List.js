import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'

import Image from 'components/image/Image'

import 'components/list/List.scss'

const rowRenderer = item => {
  if (item.type === 'hr') {
    return <hr />
  }

  const { spanClassName } = item

  return (
    <>
      <div>
        {item.image && <Image src={item.image} />}
        <h4>{item.field}</h4>
      </div>

      {!item.value ? (
        '---'
      ) : (
        <span className={spanClassName}>{item.value}</span>
      )}
    </>
  )
}

const List = ({
  id,
  keyId,
  items,
  children,
  className,
  rowClassName,
  getRowClassName,
  notFoundItem,
  childrenProps,
}) => (
  <ul className={classnames([className, 'base-list'])} id={id}>
    {items.length === 0 ? (
      <li className="base-list_not-found">{notFoundItem}</li>
    ) : (
      items.map((item, i) => (
        <li
          key={item[keyId]}
          className={classnames([rowClassName, getRowClassName(item, i)])}
        >
          {children ? children(item, childrenProps) : rowRenderer(item)}
        </li>
      ))
    )}
  </ul>
)

List.defaultProps = {
  id: null,
  keyId: 'key',
  className: '',
  rowClassName: '',
  children: '',
  getRowClassName: () => {},
  childrenProps: {},
  notFoundItem: 'Не найдено',
}

List.propTypes = {
  id: PropTypes.string,
  keyId: PropTypes.string,
  className: PropTypes.string,
  rowClassName: PropTypes.string,
  getRowClassName: PropTypes.func,
  notFoundItem: PropTypes.string,
  items: PropTypes.arrayOf(
    PropTypes.shape({
      key: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
      image: PropTypes.string,
      field: PropTypes.string,
      value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
      type: PropTypes.oneOf(['hr']),
    })
  ).isRequired,
  childrenProps: PropTypes.shape({}),
  children: PropTypes.oneOfType([PropTypes.func, PropTypes.string]),
}

export default List
