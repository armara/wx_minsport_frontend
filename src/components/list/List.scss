@import "styles/mixins";

.base-list {
  .base-list_not-found {
    font-style: italic;
  }

  hr {
    width: 100%;
    margin-top: 15px;
    margin-bottom: 15px;
  }

  li + li {
    margin-top: 15px;
  }

  li {
    display: flex;
    justify-content: space-between;

    h4 {
      @include roboto-regular-text($small, 18px, $evil-gray);
      display: flex;
      align-items: center;
    }

    & > div {
      display: flex;

      img {
        margin-right: 10px;
      }
    }

    & > span {
      @include nav-header-text($small, '18px', $black);
      display: flex;
      align-items: center;
    }
  }
}

.tabs-list {
  padding-top: 0;

  li {
    height: 45px;
  }

  li + li {
    margin-top: 0;
  }
}