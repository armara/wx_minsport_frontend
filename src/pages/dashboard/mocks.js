import React from 'react'
import dayjs from 'utils/day'

import { DAY_FORMAT } from 'utils/localization'
import ContractsTableCell from 'pages/dashboard/components/contractsTableRow/ContractsTableCell'

export const columns = [
  {
    Header: 'Организация',
    id: 'Организация',
    accessor: row => row.organizationName,
  },
  {
    Header: 'Истекает',
    id: 'Истекает',
    accessor: row => dayjs(row.finishDate).format(DAY_FORMAT),
  },
  {
    Header: 'Тип договора',
    id: 'Тип договора',
    accessor: row => (
      <ContractsTableCell
        text={row.contractTitle}
        className={
          !row.contractTitle.includes('безвозмездном')
            ? 'contracts-table-row_paid'
            : 'contracts-table-row_not-paid'
        }
      />
    ),
    style: { whiteSpace: 'unset' },
  },
]

export const data = [
  {
    foundation: 'Федерация баскетбола',
    expire: '15.01.2019',
    status: 'Оплачен',
    options: {
      isPaid: true,
    },
  },
  {
    foundation: 'Федерация баскетбола',
    expire: '12.01.2020',
    status: 'Не оплачен',
    options: {
      isPaid: false,
    },
  },
  {
    foundation: 'Федерация баскетбола',
    expire: '15.04.2020',
    status: 'Не оплачен',
    options: {
      isPaid: false,
    },
  },
  {
    foundation: 'Федерация баскетбола',
    expire: '15.10.2019',
    status: 'Оплачен',
    options: {
      isPaid: true,
    },
  },
  {
    foundation: 'Федерация баскетбола',
    expire: '09.01.2020',
    status: 'Не оплачен',
    options: {
      isPaid: false,
    },
  },
]

export const courtsList = (areas, zones) =>
  areas.map(area => ({
    title: area.name,
    rating: zones
      .filter(el => el.areaId === area.id)
      .reduce((a, b) => a + b.capacity, 0),
    key: area.id,
  }))

export const staffList = coaches =>
  coaches.map(coach => ({
    image: `data:image/jpeg;base64,${coach.avatar}`,
    field: coach.fio,
    key: coach.id,
    id: coach.id,
  }))
