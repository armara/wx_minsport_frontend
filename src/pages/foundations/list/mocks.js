import React from 'react'

import TableCellFoundation from 'components/table/tableCellIcon/TableCellIcon'
import TableCellText from 'components/table/tableCellText/TableCellText'
import TableCellActions from 'components/table/tableCellActions/TableCellActions'

export const columns = [
  {
    id: 'Организация',
    Header: 'Организация',
    isFilterable: true,
    accessor: row => <TableCellFoundation foundation={row.foundation} />,
  },
  {
    id: 'Тип организации',
    Header: 'Тип организации',
    isFilterable: true,
    accessor: row => <TableCellText text={row['foundation-type']} />,
  },
  {
    id: 'Вид организации',
    Header: 'Вид организации',
    isFilterable: true,
    accessor: row => <TableCellText text={row['organization-type']} />,
  },
  {
    id: 'Код организации',
    Header: 'Код организации',
    isFilterable: true,
    accessor: row => <TableCellText text={row['foundation-id']} />,
  },
  {
    id: 'Код / Вид спорта',
    Header: 'Код / Вид спорта',
    isFilterable: true,
    accessor: row => <TableCellText text={row['sport-id']} />,
  },
  {
    id: 'Регистрация',
    Header: 'Регистрация',
    isFilterable: true,
    accessor: row => <TableCellText text={row['registration-date']} />,
  },
  {
    id: 'Действия',
    Header: ' ',
    isFilterable: false,
    accessor: () => <TableCellActions />,
  },
]

export const data = [
  {
    foundation: 'Российская Федерация Баскетбола',
    'foundation-type': 'Федерация',
    'organization-type': 'ШОР',
    'foundation-id': '014 000 Я',
    'sport-id': '152 9013 19 С',
    'registration-date': '12.08.1976',
    actions: '',
  },
  {
    foundation: 'Российская Федерация Баскетбола',
    'foundation-type': 'Федерация',
    'organization-type': 'ШОР',
    'foundation-id': '014 001 Я',
    'sport-id': '152 9013 20 С',
    'registration-date': '12.08.1976',
    actions: '',
  },
  {
    foundation: 'Российская Федерация Баскетбола',
    'foundation-type': 'Федерация',
    'organization-type': 'ШОР',
    'foundation-id': '014 002 Я',
    'sport-id': '152 9013 21 С',
    'registration-date': '12.08.1976',
    actions: '',
  },
  {
    foundation: 'Российская Федерация Баскетбола',
    'foundation-type': 'Федерация',
    'organization-type': 'ШОР',
    'foundation-id': '014 003 Я',
    'sport-id': '152 9013 22 С',
    'registration-date': '12.08.1976',
    actions: '',
  },
]

export const dropdownItems = [
  { title: 'Бюджет', id: 'budget' },
  { title: 'Армия', id: 'forces' },
]
