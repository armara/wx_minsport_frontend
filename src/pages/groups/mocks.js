export const sportsmen = [
  {
    id: '63f268b7-9dcc-49cc-9ee8-8b4bbe84816f',
    title: 'Никита Иванович Панин',
    disciplineTitle: 'Футбол на траве',
    sportTitle: 'Футбол',
    age: '10 лет',
    img: '',
    rankName: 'МСМК',
  },
  {
    id: '63f268b7-9dcc-49cc-2ee8-8b4bbe84816f',
    title: 'Никита Иванович Панин',
    disciplineTitle: 'Футбол на траве',
    sportTitle: 'Футбол',
    age: '10 лет',
    img: '',
    rankName: 'МСМК',
  },
  {
    id: '63f268b7-9dcc-49cc-9ee8-8b46be84816f',
    title: 'Никита Иванович Панин',
    disciplineTitle: 'Футбол на траве',
    sportTitle: 'Футбол',
    age: '10 лет',
    img: '',
    rankName: 'МСМК',
  },
  {
    id: '63f268b7-9dcc-49cc-9ee8-8b43be84816f',
    title: 'Никита Иванович Панин',
    disciplineTitle: 'Футбол на траве',
    sportTitle: 'Футбол',
    age: '10 лет',
    img: '',
    rankName: 'МСМК',
  },
  {
    id: '63f268b7-9dcc-49cc-9ee8-8b4bbg84816f',
    title: 'Никита Иванович Панин',
    disciplineTitle: 'Футбол на траве',
    sportTitle: 'Футбол',
    age: '10 лет',
    img: '',
    rankName: 'МСМК',
  },
  {
    id: '63f268b7-9dcc-49cc-9ee8-8b4bbe84f16f',
    title: 'Никита Иванович Панин',
    disciplineTitle: 'Футбол на траве',
    sportTitle: 'Футбол',
    age: '10 лет',
    img: '',
    rankName: 'МСМК',
  },
  {
    id: '63f268b7-9dcc-49cc-9ee8-8s4bbe84816f',
    title: 'Никита Иванович Панин',
    disciplineTitle: 'Футбол на траве',
    sportTitle: 'Футбол',
    age: '10 лет',
    img: '',
    rankName: 'МСМК',
  },
]

export const groups = [
  {
    agesRange: '18-50 лет',
    disciplineTitle: 'Футбол на траве',
    id: '63f268b7-9dcc-49cc-9ee8-8b4bbe84816f',
    membersCount: 2,
    membersMax: 50,
    price: 100,
    sportTitle: 'Футбол',
    title: 'Футбол (дети из 48 школы)',
  },
  {
    agesRange: '18-50 лет',
    disciplineTitle: 'Футбол на траве',
    id: '63f268b7-39dcc-49cc-9ee8-8b4bbe84816f',
    membersCount: 2,
    membersMax: 50,
    price: 100,
    sportTitle: 'Футбол',
    title: 'Футбол (дети из 4 школы)',
  },
  {
    agesRange: '18-50 лет',
    disciplineTitle: 'Футбол на траве',
    id: '63f268b7-9d2c-49cc-9ee8-8b4bbe84816f',
    membersCount: 2,
    membersMax: 50,
    price: 100,
    sportTitle: 'Футбол',
    title: 'Баскетбол (дети из 17 школы)',
  },
  {
    agesRange: '18-50 лет',
    disciplineTitle: 'Футбол на траве',
    id: '63f268b7-9dcc-49cc-9ee8-8b47be84816f',
    membersCount: 2,
    membersMax: 50,
    price: 100,
    sportTitle: 'Футбол',
    title: 'Плаванье (дети из 34 школы)',
  },
]
