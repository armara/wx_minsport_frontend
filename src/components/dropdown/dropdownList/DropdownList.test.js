import React from 'react'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'

import DropdownList from 'components/dropdown/dropdownList/DropdownList'
import { dropdownItems } from 'pages/foundations/list/mocks'

configure({ adapter: new Adapter() })

describe('<DropdownList />', () => {
  it('renders correctly default props', () => {
    const tree = renderer
      .create(
        <DropdownList items={dropdownItems} titleKey="title" idKey="id" />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
