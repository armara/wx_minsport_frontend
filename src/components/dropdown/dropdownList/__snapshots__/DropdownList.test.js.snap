// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropdownList /> renders correctly default props 1`] = `
<ul
  className="dropdown-content menu-list"
>
  <button
    className="dropdown-item button base-button"
    disabled={false}
    id={null}
    onClick={[Function]}
    title=""
    type="button"
  >
    
    Бюджет
  </button>
  <button
    className="dropdown-item button base-button"
    disabled={false}
    id={null}
    onClick={[Function]}
    title=""
    type="button"
  >
    
    Армия
  </button>
</ul>
`;
